### PR TITLE
Refactor 2 - WIP

### PR DESCRIPTION
**Pull request by rogerhutchings**
So, following standup today, we've decided to:

- take the code from the current [Panoptes-Front-End `api` folder](https://github.com/zooniverse/Panoptes-Front-End/tree/master/app/api)
- rewrite it in ES5
- publish it as a new major version of this repo

This makes Panoptes-Front-End the standard, allows us to replace the API code in there, and means we have a working client for custom projects (including AnnoTate and Shakespeare's World, which already use the lib).

This will also supersede #14.

it begins

### CODE REVIEW

**Comment by brian-c** [comment]
Tests are all passing in Node now, wooo. Some changes to json-api-client were required, which I'm making in the fix-requests branch there.

I took some liberty with the non-Node-module bits: the dist/...js file is no longer included, so if someone wants a single file, they need to run the build script themselves (we should really really discourage this). And I deleted the bower file: bower is doomed, nobody should be starting a new project with it. Any objections?

Still need to try plugging it into the PFE to sure it still works in there, but the API is the same now so I _think_ it'll be cool.

**Comment by rogerhutchings** [comment]
Awesome :) Nope. Bower's done, and if people want to use it in a new project they can install this via Git. Same goes for the `dist` dir.

I want to add one more feature, which is OAuth signin (a la Wildcam Gorongosa etc). I might try doing that in a separate branch and merging it here though

**Comment by rogerhutchings** [comment]
I've also added `sugar-client` as it seems to be a dependency - via https://github.com/zooniverse/panoptes-javascript-client/blob/epic-refactor/lib/sugar.js#L1

File: lib/sugar.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  SugarClient.host = config.sugarHost;
+
+  var sugarApiClient = new JSONAPIClient(config.sugarHost, {
+    'Content-Type': 'application/json',
+    'Accept': 'application/json',
+  });
+
+  var sugarClient = new SugarClient();
+
+  auth.listen('change', function() {
```
Comment (rogerhutchings) [inline lib/sugar.js]: @brian-c auth is undefined here, and I don't think `change` is emitted anymore..?

Comment (brian-c) [inline lib/sugar.js]: Switched all those refs to `auth`, I don't really consider that module a client.

The module should still emit change events, is it not doing that somewhere?

Comment (rogerhutchings) [inline lib/sugar.js]: Fair enough. Still can't find a change event being emitted in there though...

(Happy New Year btw :) )

**Comment by brian-c** [comment]
Awaiting a version bump after zooniverse/json-api-client#24, then I think this is solid.

**Comment by brian-c** [comment]
All set now.

**Comment by rogerhutchings** [comment]
@brian-c Looks good. The only thing that I can see is that issue in `sugar.js` where it's listening for an auth change that as far as I can see isn't being emitted anywhere. Am I just misunderstanding it?

**Comment by rogerhutchings** [comment]
Just thought of another thing. Given that `json-api-client` now has superagent as a dependency, we can probably trash it here?

**Comment by brian-c** [comment]
@rogerhutchings Change events are fired whenever `update` is called on a model. Is this not happening in auth?

**Comment by rogerhutchings** [comment]
Ah - I think I was just being dumb and looking for the event to be emitted explicitly. I think this is good to go!

**Comment by srallen** [comment]
I'm seeing a few fat arrows still used in the tests:
https://github.com/zooniverse/panoptes-javascript-client/blob/epic-refactor/test/auth.js#L10
https://github.com/zooniverse/panoptes-javascript-client/blob/epic-refactor/test/auth.js#L22
https://github.com/zooniverse/panoptes-javascript-client/blob/epic-refactor/test/auth.js#L25

**Comment by brian-c** [comment]
Fixed.

**Pull request merged**